includes/service-providers/mailchimp/class-newspack-newsletters-mailchimp.php

```diff
@@ -1054,6 +1054,10 @@ public function process_link( $processed, $original ) {
 	 * @return array|WP_Error Contact data if it was added, or error otherwise.
 	 */
 	public function add_contact( $contact, $list_id = false ) {
+		// If RAS is available and a default audience is set.
+		if ( false === $list_id && class_exists( 'Newspack\Reader_Activation' ) && method_exists( 'Newspack\Data_Events\Connectors\Mailchimp', 'get_audience_id' ) ) {
+			$list_id = \Newspack\Data_Events\Connectors\Mailchimp::get_audience_id();
+		}
```

Comment:
Back to the original point of removing the line from #1355, this would make every contact subscribed to an audience and receive emails they didn't sign up for. This can result in a pretty messy database to deal with later on.

The [sync made through Mailchimp connector](https://github.com/Automattic/newspack-plugin/blob/fcad059abbe0314ba8a6f3eaaf92d5c96350e248/includes/data-events/connectors/class-mailchimp.php#L135-L162) creates the contact as transactional if it's a new or unsubscribed contact, which is the way Mailchimp allows a contact to exist without being subscribed.

Reply:
Thanks for clarifying the difference between this and the `put` method in the Mailchimp connector. As discussed offline, a quick fix (f176abb) would be to simply use this `put` method if we need to add the contact as a transactional contact. The better long-term solution would be to move all of this to the connector.

Reply:
Feels this method is doing a lot more than it should. Have you considered staying in Newspack Plugin and replace `	\Newspack_Newsletters_Subscription::add_contact( $contact );` with `\Newspack\Data_Events\Connectors\Mailchimp::put();` [here](https://github.com/Automattic/newspack-plugin/blob/fcad059abbe0314ba8a6f3eaaf92d5c96350e248/includes/reader-revenue/woocommerce/class-woocommerce-connection.php#L360)? This is remaining method that rely on `add_contact` and is used by the hooks on `woocommerce_payment_complete` and `woocommerce_subscription_status_updated`.

Reply:
Hmm, that `sync_reader_from_order` method is ESP-agnostic, so we would introduce a condition to call `put` if the ESP is MC and continue with the regular `add_contact` call for any other ESP. It does seem like it will require less code, but what's your opinion on that?

Reply:
Hm... I completely missed that...

I'm not a fan of how it's looking right now because it adds more back and forth to this logic, even though it's meant to be temporary. It goes from the plugin (hook), to newsletters (add_contact), then back to the plugin (Mailchimp::put).

Given that, I believe the cleaner temporary approach is still to leave Newspack Newsletters out of it and do a conditional check on the main plugin to use Mailchimp::put. WDYT?

Reply:
Sounds good—I'll roll it into the [other hotfix](https://github.com/Automattic/newspack-plugin/pull/2780)